### PR TITLE
test playbook generator

### PR DESCRIPTION
**Pull request by anara123**
resolves https://github.com/demisto/etc/issues/20944

### CODE REVIEW

**Comment by anara123** [comment]
the build fails not because of my changes. I am waiting for @ronykoz to finish his autocomplete PR. Then I will pull and run the build again. For now just review the code

File: demisto_sdk/main.py [hunk 1]
```diff
@@ -235,6 +236,39 @@ def lint(config, dir, **kwargs):
...
+@click.option(
+    '-o', '--outdir',
+    required=False,
+    help='Specify output directory')
+@click.option(
+    '-n', '--name',
+    required=True,
+    help='Specify test playbook name')
+@click.option(
+    '-t', '--ttype', default='integration',
```
Comment (ronykoz) [inline demisto_sdk/main.py]:
```suggestion
    '-t', '--type', default='integration',
```

File: demisto_sdk/main.py [hunk 2]
```diff
@@ -235,6 +236,39 @@ def lint(config, dir, **kwargs):
...
+    type=click.Choice(["integration", "script"]),
+    required=False,
+    help='Specify integration or script. The default is integration')
+@click.option(
+    '-c', '--commands',
+    required=False,
+    help='Full path of file containing commands, if specified those commands will appear in the test playbook with the '
+         'specified arguments')
```
Comment (ronykoz) [inline demisto_sdk/main.py]: Not sure that i get this explanation

File: demisto_sdk/test_playbook_generator/test_playbook_generator.py [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+        sys.exit(1)
+
+    generator = TestPlaybookGenerator(
+        infile=args.infile,
+        outdir=args.outdir,
+        name=args.name,
+        _type=args.type
+    )
+
+    generator.run()
```
Comment (ronykoz) [inline demisto_sdk/test_playbook_generator/test_playbook_generator.py]: remove

File: demisto_sdk/test_playbook_generator/test_playbook_generator.py [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+                        },
+                        "iscontext": True
+                    }
+                }
+            ]
+        )
+
+    conditions.append(condition)
+
+    return conditions, command_have_outputs
```
Comment (ronykoz) [inline demisto_sdk/test_playbook_generator/test_playbook_generator.py]: why all of those functions above are not within the playbook class?

File: demisto_sdk/test_playbook_generator/test_playbook_generator.py [hunk 5]
```diff
@@ -0,0 +1,340 @@
+from ruamel.yaml import YAML
+import argparse
+import json
+import os
+import sys
+
+
+class Playbook:
```
Comment (ronykoz) [inline demisto_sdk/test_playbook_generator/test_playbook_generator.py]: maybe lets call it PlaybookYaml?

File: demisto_sdk/test_playbook_generator/test_playbook_generator.py [hunk 6]
```diff
@@ -0,0 +1,340 @@
...
+    def run(self):
+        ryaml = YAML()
+        ryaml.preserve_quotes = True
+        with open(self.integration_yml_path, 'r') as yf:
+            yaml_obj = ryaml.load(yf)
+
+        commands = None
+        if self.commands_file_path:
+            with open(self.commands_file_path, 'r') as cf:
+                commands = cf.readlines()
```
Comment (ronykoz) [inline demisto_sdk/test_playbook_generator/test_playbook_generator.py]: should be `load_commands(....`

File: demisto_sdk/test_playbook_generator/test_playbook_generator.py [hunk 7]
```diff
@@ -0,0 +1,340 @@
...
+            script_name = yaml_obj.get('name')
+            task_command = create_script_task(test_playbook_v1.task_counter, script_name)
+            test_playbook_v1.add_task(task_command)
+
+            conditions, command_have_outputs = outputs_to_condition(yaml_obj.get('outputs', []))
+            if command_have_outputs:
+                task_verify_outputs = create_verify_outputs_task(test_playbook_v1.task_counter, conditions)
+                test_playbook_v1.add_task(task_verify_outputs)
+
+        test_playbook_v1.add_task(create_end_task(test_playbook_v1.task_counter))
```
Comment (ronykoz) [inline demisto_sdk/test_playbook_generator/test_playbook_generator.py]: generate_test_playbook

**Comment by ronykoz** [comment]
@anara123 Pull from master

File: CHANGELOG.md [hunk 8]
```diff
@@ -13,6 +13,13 @@
     * Playbooks
     * Integrations
 * Improved user outputs for **secrets** command.
+* Added auto-complete functionality to demisto-sdk.
+* Added git parameter in **lint** to run only on changed packages.
+* Added the **run-playbook** command
+* Added **run** command which runs a command in the Demisto playground.
+* Added **upload** command which uploads an integration or a script to a Demisto instance.
+* Fixed and issue where **validate** checked if release notes exist for new integrations and scripts.
+
```
Comment (ronykoz) [inline CHANGELOG.md]: Added **generate-test-playbook** command which generates a basic test playbook for an integration or a script.

File: demisto_sdk/main.py [hunk 9]
```diff
@@ -272,11 +287,123 @@ def format(file_type, **kwargs):
...
+    help='Specify integration or script. The default is integration')
+@click.option(
+    '--no-outputs', is_flag=True,
+    help='Skip generating verification conditions for each output contextPath. Use when you want to decide which '
+         'outputs to verify and which not')
+@click.option(
+    "-v", "--verbose", help="Verbose output for debug purposes - shows full exception stack trace", is_flag=True)
+def generate_test_playbook(**kwargs):
+    generator = TestPlaybookGenerator(**kwargs)
+    generator.run()
```
Comment (ronykoz) [inline demisto_sdk/main.py]:
```suggestion
    return generator.run()
```

File: demisto_sdk/test_playbook_generator/test_playbook_generator.py (hunk 4 above)
Comment (ronykoz) [reply demisto_sdk/test_playbook_generator/test_playbook_generator.py]: ?

File: demisto_sdk/main.py (hunk 9 above)
Comment (anara123) [reply demisto_sdk/main.py]: run dont return anything

File: CHANGELOG.md [hunk 10]
```diff
@@ -20,6 +20,7 @@
 * Added **run** command which runs a command in the Demisto playground.
 * Added **upload** command which uploads an integration or a script to a Demisto instance.
 * Fixed and issue where **validate** checked if release notes exist for new integrations and scripts.
+* Added generate-test-playbook command which generates a basic test playbook for an integration or a script.
```
Comment (ronykoz) [inline CHANGELOG.md]:
```suggestion
* Added **generate-test-playbook** command which generates a basic test playbook for an integration or a script.
```

**Pull request merged**